crates/ruff_python_semantic/src/model.rs

```diff
@@ -662,12 +662,104 @@ impl<'a> SemanticModel<'a> {
                 }
             }
 
+            // FIXME: Shouldn't this happen above where `class_variables_visible` is set?
             seen_function |= scope.kind.is_function();
         }
 
         None
     }
 
+    /// Simulates a runtime load of a given [`ast::ExprName`].
+    ///
+    /// This should not be run until after all the bindings have been visited.
+    pub fn simulate_runtime_load(&self, name: &ast::ExprName) -> Option<BindingId> {
+        let symbol = name.id.as_str();
+        let range = name.range;
+        let mut seen_function = false;
+        let mut class_variables_visible = true;
+        let mut lexicographical_lookup = true;
```

Comment:
As I understand it, the idea here is that when we are doing a "lexicographical lookup", we only accept bindings that occur in source order prior to the name node whose load we are resolving, otherwise we take the last available binding in the scope, like `lookup_symbol` above always does.

I think I can see, with the benefit of having now figured out the actual meaning from reading the code, why the name "lexicographical lookup" may have been chosen here, since it is related to "lexical" (i.e. source) ordering of bindings. But this is not a use of the term "lexicographical" that I've ever previously encountered, or can find prior art for in searching (might just be my ignorance; feel free to point me to a reference), and the name did not illuminate the intent (for me, anyway.) I think a name like `source_order_sensitive_lookup` would be clearer.